pgxn/hnsw/Makefile

```diff
@@ -12,7 +12,7 @@ REGRESS_OPTS = --inputdir=test --load-extension=hnsw
 # For auto-vectorization:
 # - GCC (needs -ftree-vectorize OR -O3) - https://gcc.gnu.org/projects/tree-ssa/vectorization.html
 PG_CFLAGS += -O3
-PG_CPPFLAGS +=  -msse4.1 -O3 -march=native -ftree-vectorize -ftree-vectorizer-verbose=0
+PG_CPPFLAGS +=  -msse4.1 -O3 -march=native -ftree-vectorize
```

Comment:
I seem to have missed this last time, but I think we should get rid of `march=native` also. It overrides `msse4.1` anyway, since the builder container probably has more modern extensions than sse4.1

Reply:
If we want to have vectorization, the right way to do it is to have separate compilation units with different levels of SIMD enabled, and at runtime detect and choose which one should be called.

Reply:
Vectorization in HNSW extension is performed using compiler intrinsic function for AVX instructions:
https://neondb.slack.com/archives/C04BLQ4LW7K/p1683877850487299
Without -march=native this instructions are not generated.
I was very surprised that this version outperforms on generated with GCC -O3 ~4 times.
Right now vectorization is supported only for x86_64 platforms.
I do not think that we are going to support larger number of platforms, so having separate compilation units seems to be overkill.

Reply:
Then I think it should be `-mavx2`. `-march=native` will allow the instruction set of the machine that's building it. If there's a mismatch, there can be problems. 

We don't necessarily need to support large number of platforms, but at least having a scalar fallback would be good.

Reply:
Yes, I agree. Using ` -msse4 -mavx2` is better than `-march-native`